service/history/historyCache.go

```diff
@@ -118,8 +118,11 @@ func (c *historyCache) getOrCreateWorkflowExecution(domainID string,
 	// This will create a closure on every request.
 	// Consider revisiting this if it causes too much GC activity
 	status := cacheNotReleased
-	releaseFunc := func() {
+	releaseFunc := func(err error) {
```

Comment:
Like the fact that you changed the signature.. that's the best way to find out all places this is being used at compile time 👍